Model/StripeKiwix.swift

```diff
@@ -0,0 +1,118 @@
+            request.httpBody = try JSONEncoder().encode(SelectedPaymentAmount(from: selectedAmount))
+            let (data, response) = try await URLSession.shared.data(for: request)
+            guard let httpResponse = response as? HTTPURLResponse,
+                  (200..<300).contains(httpResponse.statusCode) else {
+                throw StripeError.serverError
+            }
+            let json = try JSONDecoder().decode(ClientSecretKey.self, from: data)
+            return .success(json.secret)
+        } catch let serverError {
+            return .failure(serverError)
+        }
+    }
+
+    static func stripeSession(endPoint: URL) async -> PKPaymentMerchantSession? {
+        do {
+            var request = URLRequest(url: endPoint.appending(path: "payment-session"))
+            request.httpMethod = "POST"
+            request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+            let encoder = JSONEncoder()
+            encoder.keyEncodingStrategy = .convertToSnakeCase
+            request.httpBody = try encoder.encode(SessionParams())
+            let (data, response) = try await URLSession.shared.data(for: request)
+            guard let httpResponse = response as? HTTPURLResponse,
+                  (200..<300).contains(httpResponse.statusCode) else {
+                throw StripeError.serverError
+            }
+            guard let dictionary = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] else {
+                os_log("Merchant session not established: unable to decode server response", type: .debug)
+                return nil
+            }
+            return PKPaymentMerchantSession(dictionary: dictionary)
+        } catch let serverError {
+            os_log("Merchant session not established: %@", type: .debug, serverError.localizedDescription)
+            return nil
+        }
+    }
+}
+
+/// Response structure for GET {endPoint}/config
+/// {"publishable_key":"pk_test_..."}
+private struct PublishableKey: Decodable {
+    let publishableKey: String
+}
+
+/// Response structure for POST {endPoint}/create-payment-intent
+/// {"secret":"pi_..."}
+private struct ClientSecretKey: Decodable {
+    let secret: String
+}
+
+private struct SelectedPaymentAmount: Encodable {
+    let amount: Int
+    let currency: String
+
+    init(from selectedAmount: SelectedAmount) {
+        // Amount intended to be collected by this PaymentIntent.
+        // A positive integer representing how much to charge in the smallest currency unit
+        // (e.g., 100 cents to charge $1.00 or 100 to charge ¥100, a zero-decimal currency).
+        // The minimum amount is $0.50 US or equivalent in charge currency.
+        amount = Int(selectedAmount.value * 100.0)
+        currency = selectedAmount.currency
+        assert(Payment.currencyCodes.contains(currency))
+    }
+}
+private struct SessionParams: Encodable {
+    let validationUrl = "apple-pay-gateway-cert.apple.com"
```

Comment:
@rgaudin: Maybe it would be safer not to have this payload as part of the POST request.
What do you think?

Reply:
No, it's OK, that's part of the process. It's used to toggle between prod/test and global/china servers.